subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/rerun/RunOnTargetCommitCondition.kt

```diff
@@ -0,0 +1,45 @@
+package com.avito.instrumentation.rerun
+
+import com.avito.buildontarget.BuildOnTargetCommitForTestTask
+import com.avito.buildontarget.buildOnTargetTask
+import com.avito.buildontarget.hasBuildOnTargetPlugin
+import com.avito.instrumentation.configuration.InstrumentationConfiguration
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskProvider
+
+class RunOnTargetCommitCondition(private val project: Project) {
```

Comment:
Can we test this class with `ProjectBuilder`?

Reply:
added some tests on this class; i abstracted it from project